mrmustard/physics/representations.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from mrmustard import math
+from mrmustard.physics import bargmann
+from mrmustard.physics.ansatze import Ansatz, PolyExpAnsatz
+from mrmustard.utils.typing import Batch, ComplexMatrix, ComplexTensor, ComplexVector
+
+
+class Representation:
+    _contract_idxs: tuple[int, ...] = ()
+    ansatz: Ansatz
+
+    def from_ansatz(self, ansatz: Ansatz) -> Ansatz:
+        raise NotImplementedError
+
+    def __eq__(self, other):
+        return self.ansatz == other.ansatz
+
+    def __add__(self, other):
```

Comment:
I added a few docs but other are ambiguous to me. Please add them (and typehints) to the remaining ones